1_fetch.R

```diff
@@ -113,16 +114,30 @@ p1_targets_list <- list(
   # Read PRMS catchment shapefile into sf object and filter to DRB
   tar_target(
     p1_catchments_sf,
-    {st_read(dsn = p1_catchments_shp,layer="nhru", quiet=TRUE) %>%
+    sf::st_read(dsn = p1_catchments_shp,layer="nhru", quiet=TRUE) %>%
         filter(hru_segment %in% p1_reaches_sf$subsegseg) %>%
-        suppressWarnings()
-      }
+        suppressWarnings() 
   ),
   
   ## Fix issue geometries in p1_catchments_sf by defining a 0 buffer around polylines
   tar_target(
     p1_catchments_sf_valid, 
-    st_buffer(p1_catchments_sf,0)
+    sf::st_buffer(p1_catchments_sf,0)
+  ),
+  
+  # Download edited HRU polygons from https://github.com/USGS-R/drb-network-prep
+  tar_target(
+    p1_catchments_edited_gpkg,
+    download_file(GFv1_HRUs_edited_url,
+                  fileout = "1_fetch/out/GFv1_catchments_edited.gpkg", 
+                  mode = "wb", quiet = TRUE),
+    format = "file"
+  ),
+  
+  # Read in edited HRU polygons
+  tar_target(
+    p1_catchments_edited_sf,
```

Comment:
Thanks for checking that! Within `drb-network-prep` the HRUs downloaded from ScienceBase are made valid using a zero-width buffer. However, it looks like one of the catchments I made for the "split" segment 3_1 is not valid. Is that what you're seeing too?

```
> tar_load(p1_catchments_edited_sf)
> which(sf::st_is_valid(p1_catchments_edited_sf) == "FALSE")
[1] 447
> p1_catchments_edited_sf[447,]
Simple feature collection with 1 feature and 4 fields
Geometry type: MULTIPOLYGON
Dimension:     XY
Bounding box:  xmin: -74.66346 ymin: 42.1514 xmax: -74.47308 ymax: 42.31985
Geodetic CRS:  WGS 84
    PRMS_segid hru_segment hru_id hru_area_m2                           geom
447        3_1         3_1     NA   135013076 MULTIPOLYGON (((-74.59685 4...
> 

```

`sf::st_make_valid` seems to work on this polygon. We could add this step here, although I'm wondering whether you think it makes sense to implement this change in `drb-network-prep`?

Reply:
I didn't check which segments before, but yes that is what I see, too. I think it's better to change in `drb-network-prep` and update the commit link in this repo. I can do a review of the change in drb-network-prep

I'm curious: do you see my edited comment? I just noticed that it says "Pending" and I'm not sure why it wasn't included in the previous review. `Are the geometries valid in this file? using all(st_is_valid(p1_catchments_edited_sf)) returns FALSE. If they're not valid, it would probably be best to change the original gpkg file`